### PR TITLE
Add TopDownMutatorContext.CreateModule

### PR DESCRIPTION
**Pull request by colincross**
Allow a module to create other modules as if they were specified
in a blueprint file.  CreateModule takes the name of a module type,
calls the factory function the module type, and then inserts
properties into the property structs returned by the factory.

### CODE REVIEW

File: module_ctx.go [hunk 1]
```diff
@@ -737,6 +741,25 @@ func (mctx *mutatorContext) Rename(name string) {
 	mctx.rename = append(mctx.rename, rename{mctx.module.group, name})
 }
 
+// Create a new module by calling the factory method for the specified moduleType.  The module
+// will be empty, as if it were created by a blueprint file definition that contained no properties.
+// The caller should use AppendProperties
```
Comment (danw) [inline module_ctx.go]: This comment needs updating -- properties are passed in, not appended with AppendProperties

File: module_ctx.go [hunk 2]
```diff
@@ -737,6 +741,25 @@ func (mctx *mutatorContext) Rename(name string) {
...
+func (mctx *mutatorContext) CreateModule(moduleType string, props ...interface{}) {
+	module := mctx.context.newModule(moduleType)
+	if module == nil {
+		panic(fmt.Errorf("Unknown module type %q", moduleType))
```
Comment (danw) [inline module_ctx.go]: Should we take into account ignoreUnknownModuleTypes? I'm leaning towards no, but it could be useful.

**Comment by colincross** [comment]
Switch to append semantics for blueprint unpacking and CreateModule.

**Comment by colincross** [comment]
s/Switch/Switched/

**Pull request merged**